lib/mapObjects/CGTownInstance.cpp

```diff
@@ -570,7 +570,13 @@ GrowthInfo CGTownInstance::getGrowthInfo(int level) const
 	//statue-of-legion-like bonus: % to base+castle
 	TConstBonusListPtr bonuses2 = getBonuses(Selector::type()(Bonus::CREATURE_GROWTH_PERCENT));
 	for(const auto & b : *bonuses2)
-		ret.entries.push_back(GrowthInfo::Entry(b->val * (base + castleBonus) / 100, b->Description()));
+	{
+		// TODO: needs a better way to create string instead of this hack
+		const auto bonusValue = b->val;
```

Comment:
I suggest to at least make a copy of bonus just in case. Modifing const reference of such a shared thing like a bonus might yield some parallelism issues

Reply:
@dydzio0614 suggests to modify b->Description() to handle this case internally. What if we introduce just another overload Description(int bonusValue) which will ignore original bonus value?

Reply:
Or make a copy of bonus and modify value there so it is less reusable

Reply:
implemented per @dydzio0614 suggestion